services/_shared/llm_client.py

```
@@ -1135,19 +1136,29 @@ def _llmobs_tags(pr_context: Optional[PrContext]) -> dict[str, str]:
     return tags
 
 
-def _extract_usage_metrics(body: Any) -> dict[str, Optional[int]]:
+def _extract_usage_metrics(body: Any) -> dict[str, int | float]:
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `body`

(ANN401)